Tribler/Test/Core/Modules/test_versioncheck.py

```diff
@@ -0,0 +1,78 @@
+import json
+from twisted.internet.defer import Deferred
+from twisted.web import server, resource
+from Tribler.Core.Modules import versioncheck_manager
+from Tribler.Core.Utilities.network_utils import get_random_port
+from Tribler.Core.Utilities.twisted_thread import reactor, deferred
+from Tribler.Test.test_as_server import TestAsServer
+from Tribler.dispersy.util import call_on_reactor_thread
+
+
+class VersionResource(resource.Resource):
+
+    isLeaf = True
+
+    def __init__(self, response, response_code):
+        resource.Resource.__init__(self)
+        self.response = response
+        self.response_code = response_code
+
+    def render_GET(self, request):
+        request.setResponseCode(self.response_code)
+        return self.response
+
+
+class TestVersionCheck(TestAsServer):
+
+    def setUp(self):
+        self.port = get_random_port()
+        self.server = None
+        self.should_call_new_version_callback = False
+        self.new_version_called = False
+        self.test_deferred = Deferred()
+        versioncheck_manager.VERSION_CHECK_URL = 'http://localhost:%s' % self.port
+        super(TestVersionCheck, self).setUp()
+
+        self.session.notifier.notify = self.notifier_callback
+
+    def notifier_callback(self, subject, changeType, obj_id, *args):
+        self.new_version_called = True
+
+    @call_on_reactor_thread
+    def setup_version_server(self, response, response_code=200):
+        site = server.Site(VersionResource(response, response_code))
+        self.server = reactor.listenTCP(self.port, site)
+
+    def assert_new_version_called(self, _):
+        self.assertTrue(self.new_version_called == self.should_call_new_version_callback)
+        stop_deferred = self.server.stopListening()
+        stop_deferred.chainDeferred(self.test_deferred)
+
+    def start_version_checker(self):
+        self.session.lm.version_check_manager.check_new_version().addCallback(self.assert_new_version_called)
```

Comment:
Why do you not return the `self.stopListening()` Deferred (also this can return none > use a maybeDeferred!)in assert_new_version_called and chain this to the `check_newVersion()` deferred?

Then you can return `self.start_version_checker()`instead of `test_deferred` which saves using a field. @whirm likes to avoid fields in tests. The test_deferred can then be removed I think.